install-deps.sh

```diff
@@ -306,6 +306,11 @@ else
     [ $WITH_ZBD ] && with_zbd=true || with_zbd=false
     [ $WITH_PMEM ] && with_pmem=true || with_pmem=false
     source /etc/os-release
+    # check if grafonnet-lib is there else download
+    if [ ! -d "grafonnet-lib-0.1.0" ];
+    then
+        wget -c https://github.com/grafana/grafonnet-lib/archive/refs/tags/v0.1.0.tar.gz -O - | sudo tar -xz
+    fi
```

Comment:
why sudo ? i'd suggest avoid dumping code into `install-deps.sh` if unnecessary . instead i'd recommend `ExternalProject_Add()` cmake command. it took me quite a while to fix 53a5816deda0874a3a37e131e9bc22d88bb2a588 ..

Reply:
```
Wiping out workspace first.
Also:   java.nio.file.FileSystemException: /home/jenkins-build/build/workspace/ceph-api/grafonnet-lib-0.1.0/.circleci/config.yml: Operation not permitted
		at sun.nio.fs.UnixException.translateToIOException(UnixException.java:100)
		at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
		at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
```

i think this change breaks all the jenkins jobs which are scheduled by this PR.

Reply:
Addressed [here](https://github.com/ceph/ceph/pull/42194#discussion_r678416834).

I'll re-run my one-liner to delete that dir on all builders.

Reply:
@tchaikov: @aaSharma14 initially installed grafonnet as a git submodule, but I recommended him to use an alternative approach (I'm not a fan of submodules), like downloading the tarball. What's your take on that?

Reply:
BTW @tchaikov the downside of that approach you suggest is that you need internet connection during the checks phase, right? I thought that's why we wanted to prefetch all the deps...

Reply:
i think we already need internet connection during `make check`, for preparing the npm environment and for downloading nodejs packages.

Reply:
> BTW @tchaikov the downside of that approach you suggest is that you need internet connection during the checks phase, right? I thought that's why we wanted to prefetch all the deps...

might want to use "URL Download". see https://cmake.org/cmake/help/latest/module/ExternalProject.html

Reply:
Looks good @tchaikov !